olive/hardware/accelerator.py

```diff
@@ -129,3 +133,78 @@ def infer_accelerators_from_execution_provider(execution_provider: List[str]):
         if is_unique_inferring:
             return list({accelerator[0] for accelerator in accelerators})
         return None
+
+
+def create_accelerators(target: "OliveSystem", execution_providers):
+    from olive.systems.common import SystemType
+
+    if not execution_providers:
+        if target.olive_managed_env:
+            raise ValueError("Managed environment requires execution providers to be specified.")
+        elif target.system_type == SystemType.AzureML:
+            # verify the AzureML system have specified the execution providers
+            # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
+            raise ValueError("AzureMLSystem requires execution providers to be specified.")
+        elif target.system_type in (SystemType.Local, SystemType.PythonEnvironment):
+            execution_providers = target.get_supported_execution_providers()
+        elif target.system_type == SystemType.Docker:
+            # for docker system we default use CPUExecutionProvider
+            execution_providers = ["CPUExecutionProvider"]
+    logger.debug(f"Initial execution providers: {execution_providers}")
+
+    accelerators: List[str] = target.accelerators
+    if accelerators is None:
+        inferred_accelerators = AcceleratorLookup.infer_accelerators_from_execution_provider(execution_providers)
+        if not inferred_accelerators:
+            logger.warning("Cannot infer the accelerators from the target system. Use CPU as default.")
+            accelerators = ["CPU"]
+        else:
+            logger.debug(
+                f"Use inferred accelerators {inferred_accelerators} "
+                f"from given execution providers {execution_providers}."
```

Comment:
nit: use the `%` formatting for logs: https://docs.astral.sh/ruff/rules/logging-f-string/

Reply:
Thanks! Seems olive doesn't catch this kinda linter warnings. 
I updated with your suggestion. Will send separated PR to enable these linter.